From 9569c3442890cf7ca18e3fa80cd28d75bf577ad0 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 04 Mar 2021 18:42:27 +0200
Subject: [PATCH] [SW-31170] greco: remove goya2 folder and files

This patch add changes to remove goya2 folder and files as they are
renamed to greco.

Change-Id: I1c606c826e90dd553496ff68035465244cbf9935
---

diff --git a/goya2_arc/Makefile b/goya2_arc/Makefile
deleted file mode 100644
index e69de29..0000000
--- a/goya2_arc/Makefile
+++ /dev/null
diff --git a/goya2_arc/arcboot_ram.lds b/goya2_arc/arcboot_ram.lds
deleted file mode 100644
index 3630a3f..0000000
--- a/goya2_arc/arcboot_ram.lds
+++ /dev/null
@@ -1,90 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
-OUTPUT_ARCH(arc)
-
-MEMORY
-{
-	ICCM	: ORIGIN = 0x00000000, LENGTH = 64K
-	SRAM	: ORIGIN = 0x10000000, LENGTH = 128M
-	STM	: ORIGIN = 0x18000000, LENGTH = 32M
-	SPI	: ORIGIN = 0x1A000000, LENGTH = 16M
-	SBROM	: ORIGIN = 0x1B040000, LENGTH = 64K
-	FWSRAM	: ORIGIN = 0x1B050000, LENGTH = 64K
-	CS	: ORIGIN = 0x1C000000, LENGTH = 64M
-	DCCM	: ORIGIN = 0x80000000, LENGTH = 16K
-}
-
-REGION_ALIAS("REGION_TEXT",	ICCM)
-REGION_ALIAS("REGION_DATA",	DCCM)
-REGION_ALIAS("REGION_SDATA",	DCCM)
-REGION_ALIAS("REGION_SBSS",	DCCM)
-REGION_ALIAS("REGION_BSS",	DCCM)
-REGION_ALIAS("REGION_HEAP",	DCCM)
-REGION_ALIAS("REGION_STACK",	DCCM)
-
-ENTRY(__start)
-
-SECTIONS
-{
-	.text : {
-		app/crt0_ram.o (.text*)
-		*(.text .text.*)
-		_etext = . ;
-	} > REGION_TEXT
-
-	.data : AT ( LOADADDR(.text) + SIZEOF(.text) ) {
-		_data = .;
-		*(.data)
-	} > REGION_DATA
-
-	.sdata : AT ( LOADADDR(.data) + SIZEOF(.data) ) {
-		__SDATA_BEGIN__ = .;
-		*(.sdata .sdata.* .gnu.linkonce.s.*)
-		_edata = .;
-		PROVIDE (edata = .);
-	} > REGION_SDATA
-
-	.sbss : {
-		__bss_start = .;
-		PROVIDE (__sbss_start = .);
-		PROVIDE (___sbss_start = .);
-		*(.dynsbss)
-		*(.sbss .sbss.* .gnu.linkonce.sb.*)
-		*(.scommon)
-		PROVIDE (__sbss_end = .);
-		PROVIDE (___sbss_end = .);
-	} > REGION_SBSS
-
-	.bss : {
-		*(.dynbss)
-		*(.bss .bss.* .gnu.linkonce.b.*)
-		*(COMMON)
-		/* Align here to ensure that the .bss section occupies space up to
-		_end. Align after .bss to ensure correct alignment even if the
-		.bss section disappears because there are no input sections. */
-		. = ALIGN(32 / 8);
-		_end = .;
-		PROVIDE (end = .);
-	} > REGION_BSS
-
-	.heap : {
-		. = ALIGN(32 / 8);
-		__start_heap = . ;
-		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 1k)) ;
-		__end_heap = . ;
-		. = ALIGN(0x8);
-	} > REGION_HEAP
-
-	.stack : {
-		. = ALIGN(32 / 8);
-		__stack = . ;
-		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 512)) ;
-		__stack_top = . ;
-	} > REGION_STACK
-}
\ No newline at end of file
diff --git a/goya2_arc/arcboot_rom.lds b/goya2_arc/arcboot_rom.lds
deleted file mode 100644
index cce0716..0000000
--- a/goya2_arc/arcboot_rom.lds
+++ /dev/null
@@ -1,98 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-OUTPUT_FORMAT("elf32-littlearc", "elf32-bigarc", "elf32-littlearc")
-OUTPUT_ARCH(arc)
-
-MEMORY
-{
-	ICCM	: ORIGIN = 0x00000000, LENGTH = 64K
-	SRAM	: ORIGIN = 0x10000000, LENGTH = 128M
-	STM	: ORIGIN = 0x18000000, LENGTH = 32M
-	SPI	: ORIGIN = 0x1A000000, LENGTH = 16M
-	SBROM	: ORIGIN = 0x1B040000, LENGTH = 64K
-	FWSRAM	: ORIGIN = 0x1B050000, LENGTH = 64K
-	CS	: ORIGIN = 0x1C000000, LENGTH = 64M
-	DCCM	: ORIGIN = 0x80000000, LENGTH = 16K
-}
-
-REGION_ALIAS("REGION_IVT",	SBROM)
-REGION_ALIAS("REGION_TEXT",	SBROM)
-REGION_ALIAS("REGION_DATA",	DCCM)
-REGION_ALIAS("REGION_SDATA",	DCCM)
-REGION_ALIAS("REGION_SBSS",	DCCM)
-REGION_ALIAS("REGION_BSS",	DCCM)
-REGION_ALIAS("REGION_HEAP",	DCCM)
-REGION_ALIAS("REGION_STACK",	DCCM)
-
-ENTRY(__start)
-
-SECTIONS
-{
-	_dccm_begin = ORIGIN(DCCM);
-	_dccm_end = ORIGIN(DCCM) + LENGTH(DCCM);
-
-	.ivt : {
-		KEEP (*(.ivt));
-	} > REGION_IVT
-
-	.text : {
-		app/crt0_rom.o (.text*)
-		*(.text .text.*)
-		_etext = . ;
-	} > REGION_TEXT
-
-	.data : AT ( LOADADDR(.text) + SIZEOF(.text) ) {
-		_data = .;
-		*(.data)
-	} > REGION_DATA
-
-	.sdata : AT ( LOADADDR(.data) + SIZEOF(.data) ) {
-		__SDATA_BEGIN__ = .;
-		*(.sdata .sdata.* .gnu.linkonce.s.*)
-		_edata = .;
-		PROVIDE (edata = .);
-	} > REGION_SDATA
-
-	.sbss : {
-		__bss_start = .;
-		PROVIDE (__sbss_start = .);
-		PROVIDE (___sbss_start = .);
-		*(.dynsbss)
-		*(.sbss .sbss.* .gnu.linkonce.sb.*)
-		*(.scommon)
-		PROVIDE (__sbss_end = .);
-		PROVIDE (___sbss_end = .);
-	} > REGION_SBSS
-
-	.bss : {
-		*(.dynbss)
-		*(.bss .bss.* .gnu.linkonce.b.*)
-		*(COMMON)
-		/* Align here to ensure that the .bss section occupies space up to
-		_end. Align after .bss to ensure correct alignment even if the
-		.bss section disappears because there are no input sections. */
-		. = ALIGN(32 / 8);
-		_end = .;
-		PROVIDE (end = .);
-	} > REGION_BSS
-
-	.heap : {
-		. = ALIGN(32 / 8);
-		__start_heap = . ;
-		. = . + (DEFINED(__HEAP_SIZE) ? __HEAP_SIZE : (DEFINED(__DEFAULT_HEAP_SIZE) ? __DEFAULT_HEAP_SIZE : 5k)) ;
-		__end_heap = . ;
-		. = ALIGN(0x8);
-	} > REGION_HEAP
-
-	.stack : {
-		. = ALIGN(32 / 8);
-		__stack = . ;
-		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 2k)) ;
-		__stack_top = . ;
-	} > REGION_STACK
-}
\ No newline at end of file
diff --git a/goya2_arc/asic.c b/goya2_arc/asic.c
deleted file mode 100644
index 3f6ceaa..0000000
--- a/goya2_arc/asic.c
+++ /dev/null
@@ -1,6 +0,0 @@
-/*
- * Copyright (C) 2017 HabanaLabs Ltd.
- *   Author: Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
diff --git a/goya2_arc/asic_arcboot_ram.lds b/goya2_arc/asic_arcboot_ram.lds
deleted file mode 100644
index c960869..0000000
--- a/goya2_arc/asic_arcboot_ram.lds
+++ /dev/null
@@ -1,8 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
- INCLUDE arcboot_ram.lds
\ No newline at end of file
diff --git a/goya2_arc/asic_arcboot_rom.lds b/goya2_arc/asic_arcboot_rom.lds
deleted file mode 100644
index d49acc6..0000000
--- a/goya2_arc/asic_arcboot_rom.lds
+++ /dev/null
@@ -1,8 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
- INCLUDE arcboot_rom.lds
\ No newline at end of file
diff --git a/goya2_arc/dev_efuse.h b/goya2_arc/dev_efuse.h
deleted file mode 100644
index 11a630a..0000000
--- a/goya2_arc/dev_efuse.h
+++ /dev/null
@@ -1,24 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+
- *
- * Copyright (C) 2021 HabanaLabs Ltd.
- */
-
-#define EFUSE_KEY_HASH_OFFSET			0x12
-#define EFUSE_SYSCFG_ENABLE_4B_TERMINATION	BIT(25)
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION	13
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK	0x3
-#define EFUSE_SYSCFG_KEY_LOCATION		7
-#define EFUSE_SYSCFG_KEY_LOCATION_MASK		0x3
-#define EFUSE_SYSCFG_REVOKATION_LOCATION	5
-#define EFUSE_SYSCFG_REVOKATION_LOCATION_MASK	0x3
-
-#define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI	BIT(14)
-#define EFUSE_SYSCFG_KEY_LOCATION_SPI		BIT(8)
-#define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI	BIT(6)
-
-#define EFUSE_SYSCFG_DEFAULT	(EFUSE_SYSCFG_SPI_CONTROLLER | \
-				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK | \
-				 EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
-				 EFUSE_SYSCFG_KEY_LOCATION_SPI | \
-				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI | \
-				 EFUSE_SYSCFG_ENABLE_4B_TERMINATION)
diff --git a/goya2_arc/dev_security.h b/goya2_arc/dev_security.h
deleted file mode 100644
index 626ff68..0000000
--- a/goya2_arc/dev_security.h
+++ /dev/null
@@ -1,14 +0,0 @@
-// SPDX-License-Identifier:     GPL-2.0+
-/*
- * Copyright (C) 2021 HabanaLabs Ltd.
- * All Rights Reserved.
- *
- */
-
-#ifndef _GOYA2_DEV_SECURITY_H_
-#define _GOYA2_DEV_SECURITY_H_
-
-#define SECURITY_DWORD_PER_KEY				8
-#define SECURITY_KEY_HASH_COUNT				5
-
-#endif /* _GOYA2_DEV_SECURITY_H_ */
diff --git a/goya2_arc/onix.c b/goya2_arc/onix.c
deleted file mode 100644
index 50aaae6..0000000
--- a/goya2_arc/onix.c
+++ /dev/null
@@ -1,29 +0,0 @@
-/*
- * Copyright (C) 2017 HabanaLabs Ltd.
- *   Author: Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#include <unistd.h>
-#include "arcboot.h"
-#include "efuse.h"
-
-#define EFUSE_SYSCFG_ONIX_DEFAULT \
-				(EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
-				 EFUSE_SYSCFG_KEY_LOCATION_SPI | \
-				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI | \
-				 EFUSE_SYSCFG_ENABLE_VALIDATION_MASK | \
-				 EFUSE_SYSCFG_GPIO02_RECOVERY_CONTROL)
-
-int fuse_read(uint32_t word, uint32_t *val)
-{
-	int ret = -1;
-
-	if (word == EFUSE_SYSCFG_OFFSET) {
-		*val = EFUSE_SYSCFG_ONIX_DEFAULT;
-		ret = 0;
-	}
-
-	return ret;
-}
diff --git a/goya2_arc/onix_arcboot_ram.lds b/goya2_arc/onix_arcboot_ram.lds
deleted file mode 100644
index c960869..0000000
--- a/goya2_arc/onix_arcboot_ram.lds
+++ /dev/null
@@ -1,8 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
- INCLUDE arcboot_ram.lds
\ No newline at end of file
diff --git a/goya2_arc/onix_arcboot_rom.lds b/goya2_arc/onix_arcboot_rom.lds
deleted file mode 100644
index fdb13f0..0000000
--- a/goya2_arc/onix_arcboot_rom.lds
+++ /dev/null
@@ -1,13 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-INCLUDE arcboot_rom.lds
-
-MEMORY
-{
-	SBROM	: ORIGIN = 0x10000000, LENGTH = 64K
-}
diff --git a/goya2_arc/pldm.c b/goya2_arc/pldm.c
deleted file mode 100644
index 3f6ceaa..0000000
--- a/goya2_arc/pldm.c
+++ /dev/null
@@ -1,6 +0,0 @@
-/*
- * Copyright (C) 2017 HabanaLabs Ltd.
- *   Author: Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
diff --git a/goya2_arc/pldm_arcboot_ram.lds b/goya2_arc/pldm_arcboot_ram.lds
deleted file mode 100644
index c960869..0000000
--- a/goya2_arc/pldm_arcboot_ram.lds
+++ /dev/null
@@ -1,8 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
- INCLUDE arcboot_ram.lds
\ No newline at end of file
diff --git a/goya2_arc/pldm_arcboot_rom.lds b/goya2_arc/pldm_arcboot_rom.lds
deleted file mode 100644
index d49acc6..0000000
--- a/goya2_arc/pldm_arcboot_rom.lds
+++ /dev/null
@@ -1,8 +0,0 @@
-/*
- * (C) Copyright 2017 HabanaLabs, Ltd.
- * Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
- INCLUDE arcboot_rom.lds
\ No newline at end of file
diff --git a/goya2_arc/secureboot.c b/goya2_arc/secureboot.c
deleted file mode 100644
index dc7bbf8..0000000
--- a/goya2_arc/secureboot.c
+++ /dev/null
@@ -1,94 +0,0 @@
-/*
- * Copyright (C) 2017 HabanaLabs Ltd.
- *   Author: Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#include <stdio.h>
-#include "sb.h"
-#include "arcboot.h"
-#include "efuse.h"
-#include "secureboot.h"
-#include "security.h"
-
-/*
- * arcboot - Signature Verification
- *
- * A simple application which call phase0b to validate the signature
- * of a blob of data.
- *
- * Both the protected data and the signature of the protected data can be
- * changed, but the public key should be locked into the device.
- *
- * test_phase0b_data     - blob of protected data
- * test_phase0b_data_len - size of the blob
- * test_phase0b_sig      - signature of the blob
- * test_phase0b_sig_len  - signature size
- * test_public_key       - public key
- * test_public_key_len   - public key size
- */
-
-int secureboot_verify_image(uint32_t image_size, uint32_t image_offset)
-{
-	int verify = 0;
-	int err = 0;
-	int key_scan;
-	unsigned char block_buf[SECUREBOOT_HASH_BLOCK_CALC_BUF];
-	unsigned char test_phase0a_hash[SECURITY_KEY_HASH_ARRAY_BIT_SIZE];
-	uint32_t config_fuse;
-	uint16_t test_phase0b_data_len = image_size;
-	uint32_t test_phase0b_sig_len = SIG_SIZE;
-	uint32_t test_public_key_len = KEY_SIZE;
-
-	unsigned char *test_phase0b_data = (unsigned char *)(image_offset);
-	unsigned char *test_phase0b_sig = test_phase0b_data +
-					  test_phase0b_data_len;
-	unsigned char *test_public_key = test_phase0b_sig -
-					 test_public_key_len;
-	uint32_t fw_version = base_readl((uint32_t)test_phase0b_data);
-
-	/* check efuse for security validation enable */
-	config_fuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
-	if ((config_fuse & EFUSE_SYSCFG_ENABLE_VALIDATION_MASK) == 0)
-		return SECUREBOOT_VERIFY_SUCCESS;
-
-	/* check minimal valid security FW version */
-	if (security_check_valid_fw_version(fw_version) !=
-	    SECURITY_FW_VERSION_SECURE)
-		return SECUREBOOT_VERIFY_VER_FAILED;
-
-	for (key_scan = SECUREBOOT_CUSTOMER_KEY; key_scan >=
-	     SECUREBOOT_HABANA_KEY; key_scan--) {
-		if (security_get_key_hash((uint32_t *)test_phase0a_hash,
-					  key_scan)) {
-			/* check pub key hash versus efuse */
-			err = elp_secureboot_phase0a(sb_proxy_read,
-						     test_public_key,
-						     test_public_key_len,
-						     test_phase0a_hash,
-						     &verify, block_buf,
-						     sizeof(block_buf));
-		}
-		/* if key is verified we can exit the loop */
-		if (verify)
-			break;
-	}
-	/* if for loop doesn't find a valid key hash in fuse, return error */
-	if (!verify)
-		return SECUREBOOT_VERIFY_KEY_FAILED;
-
-	/* Phase0b uses the public key (test_public_key) to validate signature
-	 * (test_phase0b_sig) of the data blob (test_phase0b_data)
-	 */
-	err = elp_secureboot_phase0b(sb_proxy_read, test_phase0b_data,
-				     test_phase0b_data_len, test_public_key,
-				     test_public_key_len, test_phase0b_sig,
-				     test_phase0b_sig_len, &verify,
-				     block_buf, sizeof(block_buf));
-
-	if (err || !verify)
-		return SECUREBOOT_VERIFY_FAILED;
-
-	return verify;
-}
diff --git a/goya2_arc/security.c b/goya2_arc/security.c
deleted file mode 100644
index 1ac70fe..0000000
--- a/goya2_arc/security.c
+++ /dev/null
@@ -1,496 +0,0 @@
-/*
- * Copyright 2016-2019 HabanaLabs, Ltd.
- * All Rights Reserved.
- *
- * SPDX-License-Identifier:	GPL-2.0+
- */
-
-#include "arcboot.h"
-#include "security.h"
-#include "efuse.h"
-#include "security.h"
-#include "spi_flash.h"
-
-static void security_disable_i2c_s_mstr_if(void)
-{
-	/* set AXUSER to shared */
-	regmap_writel(I2C_S_MSTR_IF_AXUSER_CORE_SHARED_MASK,
-	      mmI2C_S_MSTR_IF_AXUSER_CORE);
-	/* override AXI prot bit by master IF to secure, unprivlaged access */
-	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
-		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	/* enable access to PSOC_SECURITY only */
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-}
-
-static void security_disable_smi_mstr_if(void)
-{
-	regmap_writel(SMI_MSTR_IF_AXUSER_CORE_SHARED_MASK,
-	      mmSMI_MSTR_IF_AXUSER_CORE);
-	/* override AXI prot bit by master IF to secure, unprivlaged access */
-	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
-		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	/* enable access to PSOC_SECURITY only */
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-}
-
-void security_disable_jtag_mstr_if(void)
-{
-	regmap_writel(JT_MSTR_IF_AXUSER_CORE_SHARED_MASK,
-		      mmJT_MSTR_IF_AXUSER_CORE);
-
-	/* override AXI prot bit by master IF to secure, unprivlaged access */
-	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
-		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	/* enable access to PSOC_SECURITY only */
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-}
-
-void security_enable_jtag_mstr_if(void)
-{
-	regmap_writel(0, mmJT_MSTR_IF_AXUSER_CORE);
-	/* don't override AXI prot bit by master IF */
-	regmap_writel(0, mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-
-	/* enable access to all */
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-}
-
-void security_disable_twi_if(void)
-{
-	security_disable_i2c_s_mstr_if();
-	security_disable_smi_mstr_if();
-}
-
-void security_jtag_dap(uint8_t status)
-{
-	/* close dap WA jira: H5-1803 */
-	regmap_writel(status, mmPSOC_CS_TRACE_SPECIAL_GLBL_SPARE_0);
-}
-
-void security_debug_if_shared(void)
-{
-	/* Configure SMI,I2C and JTAG to be "shared" engine */
-	regmap_writel(I2C_S_MSTR_IF_AXUSER_CORE_SHARED_MASK,
-		      mmI2C_S_MSTR_IF_AXUSER_CORE);
-	regmap_writel(SMI_MSTR_IF_AXUSER_CORE_SHARED_MASK,
-		      mmSMI_MSTR_IF_AXUSER_CORE);
-	regmap_writel(JT_MSTR_IF_AXUSER_CORE_SHARED_MASK,
-		      mmJT_MSTR_IF_AXUSER_CORE);
-	regmap_writel(MSTR_IF_AXUSER_CORE_SHARED,
-		      mmPSOC_MSTR_IF_AXUSER_CORE);
-}
-
-static void security_dcon_rr_lbw_config(uint8_t spi_controller_en)
-{
-	regmap_writel(SECURITY_PSOC_START_LBW,
-		mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0);
-
-	if (spi_controller_en) {
-		regmap_writel(SECURITY_QSPI_START_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
-		regmap_writel(SECURITY_QSPI_END_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
-		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
-		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_2);
-		regmap_writel(SECURITY_PSOC_END_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_2);
-	} else {
-		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
-		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
-		regmap_writel(SECURITY_PSOC_END_LBW,
-		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
-	}
-}
-
-static void security_dcon_if0_rr_hbw_config(uint8_t spi_controller_en)
-{
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-
-	if (spi_controller_en) {
-		regmap_writel(SECURITY_QSPI_START_HBW,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_QSPI_END_HBW,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	} else {
-		regmap_writel(SECURITY_FLASH_START_HBW,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_FLASH_END_HBW,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	}
-
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-}
-
-static void security_dcon_if1_rr_hbw_config(uint8_t spi_controller_en)
-{
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-
-	if (spi_controller_en) {
-		regmap_writel(SECURITY_QSPI_START_HBW,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_QSPI_END_HBW,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	} else {
-		regmap_writel(SECURITY_FLASH_START_HBW,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_FLASH_END_HBW,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	}
-
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-}
-
-void security_i2c_s_rr_hbw_config(uint8_t spi_controller_en)
-{
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-
-	if (spi_controller_en) {
-		regmap_writel(SECURITY_QSPI_START_HBW,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_QSPI_END_HBW,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	} else {
-		regmap_writel(SECURITY_FLASH_START_HBW,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_FLASH_END_HBW,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	}
-
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
-}
-
-static void security_elbi_rr_lbw_config(
-				uint8_t spi_controller_en)
-{
-	regmap_writel(SECURITY_PSOC_START_LBW,
-		mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0);
-
-	if (spi_controller_en) {
-		regmap_writel(SECURITY_QSPI_START_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
-		regmap_writel(SECURITY_QSPI_END_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
-		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
-		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_2);
-		regmap_writel(SECURITY_PSOC_END_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_2);
-	} else {
-		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
-		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
-		regmap_writel(SECURITY_PSOC_END_LBW,
-			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
-	}
-}
-
-void security_smi_rr_hbw_config(uint8_t spi_controller_en)
-{
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-
-	if (spi_controller_en) {
-		regmap_writel(SECURITY_QSPI_START_HBW,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_QSPI_END_HBW,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	} else {
-		regmap_writel(SECURITY_FLASH_START_HBW,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_FLASH_END_HBW,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	}
-
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
-}
-
-void security_jtag_rr_hbw_config(uint8_t spi_controller_en)
-{
-	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-
-	if (spi_controller_en) {
-		regmap_writel(SECURITY_QSPI_START_HBW,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_QSPI_END_HBW,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	} else {
-		regmap_writel(SECURITY_FLASH_START_HBW,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(SECURITY_FLASH_END_HBW,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(SECURITY_HBW_MSB_ZERO,
-			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	}
-
-	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(SECURITY_HBW_MSB_ZERO,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
-	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
-	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
-		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
-}
-
-void security_pcie_rr_config(uint8_t spi_en)
-{
-	regmap_writel(PCIE_WRAP_PCIE_PROT_OVR_WR_EN_MASK |
-		      PCIE_WRAP_PCIE_PROT_OVR_RD_EN_MASK,
-		      mmPCIE_WRAP_PCIE_PROT_OVR);
-	regmap_writel(PCIE_MSTR_RR_MSTR_IF_AXUSER_CORE_SHARED_MASK,
-		      mmPCIE_MSTR_RR_MSTR_IF_AXUSER_CORE);
-	security_elbi_rr_lbw_config(spi_en);
-	security_dcon_rr_lbw_config(spi_en);
-	security_dcon_if0_rr_hbw_config(spi_en);
-	security_dcon_if1_rr_hbw_config(spi_en);
-}
-
-uint32_t security_key_soc_location_offset(uint32_t hash_key_index)
-{
-	uint32_t ret;
-	uint8_t key_soc_base_index;
-	uint8_t key_soc_location_index;
-	uint32_t key_soc_location[] = {
-		mmPSOC_SECURITY_HABANA_KEY_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_0_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_1_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_2_0,
-		mmPSOC_SECURITY_PUBLIC_KEY_3_0
-	};
-
-	/* select which index to choose (haban, customer 0-4) */
-	key_soc_base_index = hash_key_index / SECURITY_DWORD_PER_KEY;
-	/* key is 256bit, 32bit * 8... which of the 8 location is requested */
-	key_soc_location_index = hash_key_index % SECURITY_DWORD_PER_KEY;
-
-	/* get base address of selected key index */
-	ret = key_soc_location[key_soc_base_index];
-	/* add to base address of key the spicifice location of currect DWORD */
-	ret += (key_soc_location_index * sizeof(uint32_t));
-	return ret;
-}
-
-void security_load_keys(void)
-{
-	uint8_t hash_key_index;
-	uint32_t key_hash[SECURITY_KEY_HASH_ARRAY_SIZE];
-	enum security_location key_location;
-
-	key_location = ((arcboot_rom_global.config_fuse >>
-			EFUSE_SYSCFG_KEY_LOCATION) &
-			EFUSE_SYSCFG_KEY_LOCATION_MASK);
-
-	arcboot_rom_status_update(PROGRESS_KEY_LOC);
-	arcboot_rom_status_update_location(PROGRESS_KEY_LOC_0, key_location);
-
-	switch (key_location) {
-	case SPI_FLASH:
-		spi_flash_get_hash(key_hash);
-		break;
-	case SPI_OTP:
-		spi_flash_otp_access(FLASH_OTP_ENABLE);
-		spi_flash_otp_get_hash(key_hash);
-		spi_flash_otp_access(FLASH_OTP_DISABLE);
-		break;
-	case ASIC_EFUSE:
-	default:
-		efuse_get_hash(key_hash);
-		break;
-	}
-
-	for (hash_key_index = 0; hash_key_index < SECURITY_KEY_HASH_ARRAY_SIZE;
-	     hash_key_index++) {
-		/* store habana key hash in PSOC_SECURITY */
-		regmap_writel(key_hash[hash_key_index],
-			      security_key_soc_location_offset(hash_key_index));
-		/* store key hashes in Scratch PAD Sram */
-		sp_sram_writel(key_hash[hash_key_index],
-			       SECURITY_EFUSE_KEYS_OFFSET + (hash_key_index *
-			       SECURITY_BYTE_PER_DWORD));
-	}
-}
-
-void check_and_enable_4b_termination(void)
-{
-	uint32_t spare0;
-	/*
-	 * check and enable 4B termination
-	 * set bit[1] of mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0 register
-	 * to enable LBW 4B termination.
-	 */
-	if (arcboot_rom_global.config_fuse &
-				EFUSE_SYSCFG_ENABLE_4B_TERMINATION) {
-
-		spare0 = regmap_readl(mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
-		spare0 |= BIT(1);
-		regmap_writel(spare0, mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
-	}
-}
-
-void security_handle_twi(void)
-{
-	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_DISABLE_TWI_IF_MASK)
-		security_disable_twi_if();
-}
-
-void security_disable_jtag_if(void)
-{
-	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_0);
-	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_1);
-	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
-	security_disable_jtag_mstr_if();
-}
-
-void security_enable_jtag_if(void)
-{
-	regmap_writel(0xFFFFFFFF, mmPSOC_CS_TRACE_TOP_ROM_VALID_0);
-	regmap_writel(0xFFFFFFFF, mmPSOC_CS_TRACE_TOP_ROM_VALID_1);
-	security_jtag_dap(SECURITY_JTAG_DAP_ENABLE);
-	security_enable_jtag_mstr_if();
-}
